tempodb/backend/local/local.go

```diff
@@ -104,17 +104,14 @@ func (rw *Backend) CloseAppend(ctx context.Context, tracker backend.AppendTracke
 // List implements backend.Reader
 func (rw *Backend) List(ctx context.Context, keypath backend.KeyPath) ([]string, error) {
 	path := rw.rootPath(keypath)
-	folders, err := os.ReadDir(path)
+	dirEntries, err := os.ReadDir(path)
 	if err != nil {
 		return nil, err
 	}
 
-	objects := make([]string, 0, len(folders))
-	for _, f := range folders {
-		if !f.IsDir() {
-			continue
-		}
-		objects = append(objects, f.Name())
+	objects := make([]string, 0, len(dirEntries))
+	for _, d := range dirEntries {
+		objects = append(objects, d.Name())
```

Comment:
This was changed to support the `Has()` method? I'm concerned about the impact on other uses of `List()`

Reply:
Yes. From the signature alone it is not clear whether `List` is supposed to return only directory names. The code where `List` is used ([raw.go#L158](https://github.com/grafana/tempo/blob/main/tempodb/backend/raw.go#L158) or [raw.go#L139](https://github.com/grafana/tempo/blob/main/tempodb/backend/raw.go#L139)) seems to anticipate that filenames are listed too. However, the s3 backend also only returns directories / prefixes.